pages/_app.tsx

```diff
@@ -1,14 +1,61 @@
 import 'styles/global.css'
 import 'styles/preflight.css'
+import Head from 'next/head'
 
 import { ThemeProvider } from '@mui/material'
 import { AppProps } from 'next/app'
 import { theme } from 'theme/theme'
+import { useEffect, useState } from 'react'
+import { usePathname, useSearchParams } from 'next/navigation'
+import { dasProjectsService } from 'pages/api/ProjectsService'
+
+const DEFAULT_TAG =
+  'Free tech solutions for Puget Sound nonprofits | Digital Aid Seattle'
+const TAG_NAMES = {
+  events: 'Events | Digital Aid Seattle',
+  volunteers: 'Volunteer with us | Digital Aid Seattle',
+  partners: 'Get help with your nonprofit | Digital Aid Seattle',
+  about: 'About us | Digital Aid Seattle',
+  projects: 'Projects | Digital Aid Seattle',
+  privacy: 'Privacy policy | Digital Aid Seattle',
+}
 
 export default function App({ Component, pageProps }: AppProps) {
+  const pathName = usePathname()
+  const searchParams = useSearchParams()
+
+  let pageName = null
```

Comment:
Just a recommendation, but to limit the scope of variables, `pageName` should be moved within the lookupTitle function.   the useEffect hook would depend on the pathName.

Limiting the scope of variables is a good engineering practice.

Reply:
Thank you, Jeff, for the feedback. I've refactored the code and tested it locally.